sklearn/metrics/_classification.py

```diff
@@ -2476,16 +2477,17 @@ def brier_score_loss(y_true, y_prob, sample_weight=None, pos_label=None):
     if y_prob.min() < 0:
         raise ValueError("y_prob contains values less than 0.")
 
-    # if pos_label=None, when y_true is in {-1, 1} or {0, 1},
-    # pos_label is set to 1 (consistent with precision_recall_curve/roc_curve),
-    # otherwise pos_label is set to the greater label
-    # (different from precision_recall_curve/roc_curve,
-    # the purpose is to keep backward compatibility).
     if pos_label is None:
-        if (np.array_equal(labels, [0]) or
+        if labels.dtype.kind == 'U':
```

Comment:
Not sure whether it's the correct way. We can't use _determine_key_type here, because y_true can be floats (e.g., [0.0, 1.0]) and _determine_key_type will raise an error in this case.